### ISSUE TITLE
Import is slow

### ISSUE THREAD

**Issue by tom93** [FossifyOrg/Messages#143]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

Android 8.1, Android 14

### Affected device model

Moto G5 Plus, emulator

### How did you install the app?

GitHub releases

### Steps to reproduce the bug

Import a .json backup with thousands of messages ([example](https://github.com/FossifyOrg/Messages/files/14721578/messages.json) with 3000 messages, generated using the code at the end).

### Expected behavior

The import should either complete quickly or display proper progress indication (for example [as a notification](https://developer.android.com/develop/ui/views/notifications/build-notification#progressbar)).

### Actual behavior

There is a toast saying "Importing...", and then it disappears and it seems like nothing is happening. A long time later there is a toast saying "Importing successful".

### Additional information

Originally reported by @Abcd1234-dot in https://github.com/FossifyOrg/Messages/issues/88#issuecomment-2002326219.

---

I tried making the import faster by switching from ContentProvider.insert() to ContentProvider.bulkInsert() as well as some other improvements (branch "import-bulk", [commit](https://github.com/FossifyOrg/Messages/commit/d3c73d77f9ad7fa523062bdca4d808e964826675)), however that doesn't help that much. It reduces the time by 36% on Android 8.1 (66 sec => 42 sec) and 60% on Android 14 (72 sec => 26 sec) in an emulator with the attached 3000-message example. The patched version is super fast on messages that already exist, but that doesn't help during the initial import. I'm not sure if it's worth merging, because it's a little complicated for not-very-big gains and the error handling isn't great (if one message fails, the entire chunk may be skipped).

I don't think we can make the import any faster than that, because almost all of the time is spent in the bulkInsert() call waiting for the Telephony provider, which is outside of our control. We'd have to file a bug report against Android and hope that Google the code faster.

So I guess we have to show a proper progress indicator (e.g. using a notification; or at the bare minimum show multiple toasts).

---

For testing, the following Python code can be used to generate a large backups:

```python
print("[")
for i in range(3000):
    print('%s{"subscriptionId":1,"address":"6505550","body":"Message %d","date":%d,"dateSent":0,"locked":0,"protocol":null,"read":1,"status":-1,"type":%d,"serviceCenter":null,"backupType":"sms"}' %
          ("," if i != 0 else "", i + 1, 1700000000000 + i * 60*1000, i % 2 + 1))
print("]")
```

**Comment by 4fosaefp** [comment]
I just did an import and it took a solid 5 minutes to complete. Some kind of progress indication would be really helpful.

**Comment by naveensingh** [comment]
There is a progress bar in the latest versions but it's not very useful for tracking progress:

![Image](https://github.com/user-attachments/assets/1d66b7d1-ece2-4c62-9aac-b3d5d87baafa)

**Comment by 4fosaefp** [comment]
Yeah I did see the progress bar but it just runs indefinitely, and I can't tell if the import is still going or if it hangs, especially after a couple of minute.

**Comment by NanNam226** [comment]
> Yeah I did see the progress bar but it just runs indefinitely, and I can't tell if the import is still going or if it hangs, especially after a couple of minute.

This is happening for me as well. Though in my case, the activity bar stops moving after a while, and I never get a confirmation for the transfer. 

Also, not all og my messages gets transferred. I cannot confirm that this is not an issue with export, but it seems consistent with the activity indicator stopping. 

The activity indicator stops after the screentime times out. When I turn on the screen again, the activitybar stopped moving. It happens even when I use a screen timeout of 10 minutes. Either way, the import takes way too long, and it is definitely problematic that it fails, even if it might succed with a longer screentime. The app has all the permissions it asks for, I have set battery settings to unrestrictive. 

I both export and import using fossify Messages.
